### PR TITLE
added authentication for private hf models

### LINKED ISSUE #9842: Authentication for private HF Models
We need to provide an option for authentication when private hf models are used. We can re-use the `use_auth_token` pipeline. For more info, check [hf docs](https://huggingface.co/docs/transformers/v4.25.1/en/main_classes/pipelines#transformers.pipeline.use_auth_token).

## Resources :bulb: 

:checkered_flag: To get started, please refer to the following resources:

* :books:  [Hugging Face docs](https://docs.mindsdb.com/integrations/ai-engines/huggingface)
* :point_right: [Hugging Face Handler](https://github.com/mindsdb/mindsdb/tree/main/mindsdb/integrations/handlers/huggingface_handlerr)

## :loudspeaker: Additional rewards :trophy:
This issue brings :eight:  :zero:  : points for SWAG and entry into the draw for a  :point_down: :
- :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)
:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left:

### PR DESCRIPTION
**Pull request by the-dg04**
## Description

Added authentication for private hf models by adding an optional `use_auth_token` param while creating a new hf model using `CREATE MODEL` + updated docs for the same.

Fixes #9842 

## Type of change

(Please delete options that are not relevant)

- [x] ⚡ New feature (non-breaking change which adds functionality)

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Review by ZoranPandovski — APPROVED** [review]

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb/blob/main/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by the-dg04** [comment]
I have read the CLA Document and I hereby sign the CLA

File: mindsdb/integrations/handlers/huggingface_handler/huggingface_handler.py [hunk 1]
```diff
@@ -116,14 +120,14 @@ def create(self, target, args=None, **kwargs):
...
-                                             tokenizer=hf_model_storage_path)
+                                             tokenizer=hf_model_storage_path,use_auth_token=use_auth_token)
             logger.debug(f'Model already downloaded!')
         ####
         # Otherwise download it
         except (ValueError, OSError):
             try:
                 logger.debug(f"Downloading {model_name}...")
-                pipeline = transformers.pipeline(task=args['task_proper'], model=model_name)
+                pipeline = transformers.pipeline(task=args['task_proper'], model=model_name,use_auth_token=use_auth_token)
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/huggingface_handler/huggingface_handler.py:130]: Let's not do this. Let's only accept it as a boolean argument.

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]
Hi @the-dg04,
According to this [post](https://discuss.huggingface.co/t/the-use-auth-token-argument-is-deprecated-and-will-be-removed-in-v5-of-transformers/53943) in the HuggingFace forum, the `use_auth_token` parameter is going to be depreciated in the near future. Can we please use the `token` parameter instead?

Furthermore, please do the following:
1. Post a screenshot or video of successfully executing queries via the MinsdsDB Editor after your changes.
2. Add these changes to the documentation for the integration in addition to the README.

**Comment by ZoranPandovski** [comment]
Hey @the-dg04  we haven't heard back regarding this PR . Just a friendly reminder that if there's no response, we may need to close this PR to keep the repository organized. If you're still working on this or would like to continue, please let us know! Even if we close it, you're always welcome to comment here, and we can reopen it at any time

**Pull request closed without merging**